Editor/Properties/TeleportPropertyEditor.cs

```diff
@@ -0,0 +1,95 @@
+using UnityEngine.XR.Interaction.Toolkit;
+
+namespace Innoactive.CreatorEditor.XRInteraction
+{
+    /// <summary>
+    /// Custom inspector for <see cref="TeleportProperty"/>, adding a button to automatically configure <see cref="Innoactive.Creator.XRInteraction.TeleportationAnchor"/>s.
+    /// </summary>
+    [CustomEditor(typeof(TeleportProperty)), CanEditMultipleObjects]
+    internal class TeleportPropertyEditor : Editor
+    {
+        private bool isNotSetup = true;
+        private const string AnchorPrefabName = "anchor";
+
+        private void OnEnable()
+        {
+            TeleportProperty teleportProperty = target as TeleportProperty;
+            TeleportationAnchor teleportAnchor = teleportProperty.GetComponent<TeleportationAnchor>();
+            
+            if (teleportProperty.transform.childCount != 0 && teleportAnchor.teleportAnchorTransform.name == AnchorPrefabName)
+            {
+                isNotSetup = false;
+            }
+        }
+
+        public override void OnInspectorGUI()
+        {
+            base.OnInspectorGUI();
+            EditorGUILayout.Space();
+
+            if (isNotSetup && GUILayout.Button("Set Default Teleportation Anchor"))
+            {
+                foreach (UnityEngine.Object targetObject in serializedObject.targetObjects)
+                {
+                    if (targetObject is TeleportProperty teleportationAnchor)
+                    {
+                        ConfigureDefaultTeleportationAnchor(teleportationAnchor);
+                    }
+                }
+            }
+        }
+        
+        private  void ConfigureDefaultTeleportationAnchor(TeleportProperty teleportationAnchor)
+        {
+            TeleportationAnchor teleportAnchor = teleportationAnchor.GetComponent<TeleportationAnchor>();
+            Transform anchorTransform = teleportationAnchor.transform;
+
+            try
+            {
+                GameObject anchorPrefab = Instantiate(Resources.Load<GameObject>(AnchorPrefabName));
+                anchorPrefab.name = anchorPrefab.name.Remove(AnchorPrefabName.Length);
+                
+                anchorPrefab.transform.SetPositionAndRotation((anchorTransform.position + (Vector3.up * 0.01f)), anchorTransform.rotation);
+                anchorPrefab.transform.SetParent(anchorTransform);
+
+                teleportAnchor.teleportAnchorTransform = anchorPrefab.transform;
+
+                string teleportLayerName = "XR Teleport";
+                LayerMask teleportLayer = LayerMask.NameToLayer(teleportLayerName);
+                teleportationAnchor.gameObject.layer = teleportLayer;
+                teleportAnchor.interactionLayerMask = 1 <<teleportLayer;
+
+                teleportAnchor.customReticle = Resources.Load<GameObject>("teleportReticle");
+                teleportAnchor.matchOrientation = MatchOrientation.TargetUpAndForward;
+
+                Bounds bounds = new Bounds(Vector3.zero, Vector3.zero);
```

Comment:
Method feels too long, maybe split setup itself in some smaller private methods which makes it easier to understand what is happening here.